toptal/codeowners-checker
# [BT-313] Guess GitHub organization and add configuration to override it

Instead of relying on setting `ENV["GITHUB_ORGANIZATION"]`, the organization is now automatically inferred based on the remote origin URL. For example, given `git@github.com:toptal/codeowners-checker.git` as the origin URL, the guessed organization will be `toptal`. It supports remote origin URLs in both git and HTTP(S) formats from Github.

An option to override the default guessed organization is also present and can be used by invoking `codeowners-checker config organization <organization>`, and you can check the value set with `codeowners-checker config list`

A deprecation warning has also been added to warn that `ENV["GITHUB_ORGANIZATION"]` is deprecated.

## HOW TO TEST
download branch, install it with `bundle exec rake install`.

Run `codeowners-checker config list` on some repository. You should see a message saying `default_organization: "<something>"`. 

You can also `export GITHUB_ORGANIZATION=something` and verify that the warning message appears when running the previous command, but doesn't when it's set to blank

If you can see a default_organization value (it's not ""), then when running `codeowners-checker check` it should not ask for the organization, even when ENV["GITHUB_ORGANIZATION"] is not set.

You can also test that `codeowners-checker config organization new-org-name` changes the value shown by `codeowners-checker config list`

Finally, when setting the organization correctly on a project and running `codeowners-checker fetch` it should download an `OWNERS` file with the correct organization teams/contributors.

The test coverage went up from 96.88% to 98.34% and all tests should be green (tests can be run with `bundle exec rake spec`)
